astropy/modeling/fitting.py

```diff
@@ -1218,14 +1218,17 @@ def _wrap_deriv(params, model, weights, x, y, z=None):
                         output = np.array(
                             [np.ravel(_) for _ in np.atleast_2d(weights).T * fit_deriv]
                         )
```

Comment:
@Shaheer-Ahmd @WilliamJamieson one of the `output=` can be moved out of the try block, either before or to the `else` block.